blog/content/post/2024-04-11-aem_phase4.md

```diff
@@ -0,0 +1,257 @@
+| DCE Preamble                                     | n/a                                   | n/a                  | GRUB2 - software that prepares and initializes DL_Event    |
+| DLME (Dynamically Launched Measured Environment) | MLE (Measured Launch Environment)     | SK (Security Kernel) | Xen, DLME definition also covers dom0 kernel and initramfs |
+
+Apart from addition of SKL and supporting it both in GRUB2 as well as in Xen,
+few other, usually smaller changes were made:
+
+- Script for dumping DRTM TPM event log was enhanced to support event log header
+  as defined by TCG for TPM1.2 before the one specific to TXT. SKL puts TXT
+  header as part of vendor info in TCG header, this gives a good compromise
+  between two very similar formats - entries are identical, but headers are not.
+- Scriptlet that creates grub.cfg file now searches for SKL, in addition to
+  SINIT ACMs.
+- GRUB2 is now able to choose best `slaunch_module` from all passed ones. This
+  should allow users to put all ACMs and SKL in `/boot` directory, without
+  having to choose the proper one.
+- This release is published in a form of repository to (hopefully) make
+  installation process easier.
+
+List of pull requests created and merged as part of this phase:
+
+- [GRUB #17](https://github.com/TrenchBoot/grub/pull/17)
+- [GRUB #18](https://github.com/TrenchBoot/grub/pull/18)
+- [GRUB #19](https://github.com/TrenchBoot/grub/pull/19)
+- [Xen #9](https://github.com/TrenchBoot/xen/pull/9)
+- [Xen #10](https://github.com/TrenchBoot/xen/pull/10)
+- [Xen #12](https://github.com/TrenchBoot/xen/pull/12)
+- [Xen #13](https://github.com/TrenchBoot/xen/pull/13)
+- [Secure Kernel Loader #14](https://github.com/TrenchBoot/secure-kernel-loader/pull/14)
+- [Secure Kernel Loader #16](https://github.com/TrenchBoot/secure-kernel-loader/pull/16)
+- [Secure Kernel Loader #17](https://github.com/TrenchBoot/secure-kernel-loader/pull/17)
+- [Secure Kernel Loader #18](https://github.com/TrenchBoot/secure-kernel-loader/pull/18)
+- [Secure Kernel Loader #19](https://github.com/TrenchBoot/secure-kernel-loader/pull/19)
+- [Qubes Anti Evil Maid #9](https://github.com/TrenchBoot/qubes-antievilmaid/pull/9)
+- [Qubes Anti Evil Maid #10](https://github.com/TrenchBoot/qubes-antievilmaid/pull/10)
+- [Qubes Anti Evil Maid #11](https://github.com/TrenchBoot/qubes-antievilmaid/pull/11)
+
+## Installation
+
+To install, you have to first add a new repository and import a public part of
+a key that was used to sign RPM packages. To add a new repository, create in
+dom0 as root `/etc/yum.repos.d/aem.repo` with the following content:
+
+```text
+[aem]
+name = Anti Evil Maid based on TrenchBoot
+baseurl = https://dl.3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_v0.4
+gpgcheck = 1
+gpgkey = https://dl.3mdeb.com/open-source-firmware/QubesOS/trenchboot_aem_v0.4/RPM-GPG-KEY-aem
```

Comment:
URLs will change!

Reply:
Updated in https://github.com/3mdeb/news-and-ideas/pull/592/commits/a6ff9041f440175591c72e7cc7c9f4ab7706411e and https://github.com/3mdeb/news-and-ideas/pull/592/commits/f7b994439be0183065e0a322e8aa202dcf2219b9